projects/spectator/src/lib/spectator/create-factory.ts

```diff
@@ -55,14 +55,31 @@ export function overrideComponentIfProviderOverridesSpecified<C>(options: Requir
  */
 export function overrideModules(options: Required<BaseSpectatorOptions>): void {
   if (options.overrideModules.length) {
-    options.overrideModules.forEach(overrideModule => {
+    options.overrideModules.forEach((overrideModule) => {
       const [ngModule, override] = overrideModule;
 
       TestBed.overrideModule(ngModule, override);
     });
   }
 }
 
+/**
+ * @internal
+ */
+export function overrideComponents(options: Required<BaseSpectatorOptions>): void {
+  if (options.overrideComponents.length) {
+    options.overrideComponents.forEach((overrideComponent) => {
+      const [component, override] = overrideComponent;
+
+      if (!ɵisStandalone(component)) {
```

Comment:
There is no need to use an internal method. https://netbasal.com/a-new-method-for-retrieving-metadata-about-angular-components-b494351dd130